src/backend/cuda/CMakeLists.txt

```diff
@@ -334,17 +334,23 @@ SET(libdevice_bc "")
 IF (USE_LIBDEVICE)
   SET(libdevice_computes "")
   LIST(APPEND libdevice_computes "20" "30" "35" "50")
+  LIST(APPEND cuda_root_list "${CUDA_NVVM_HOME}")
+  # Below location of libdevice files when installed via package manager on Ubuntu 16.04
+  LIST(APPEND cuda_root_list "/usr/lib/nvidia-cuda-toolkit")
   FOREACH(libdevice_compute ${libdevice_computes})
-    SET(_libdevice_bc_file "${CUDA_NVVM_HOME}/libdevice/libdevice.compute_${libdevice_compute}.10.bc")
-    SET(_libdevice_bc_copy "${PROJECT_BINARY_DIR}/src/backend/cuda/compute_${libdevice_compute}.bc")
-    IF (EXISTS ${_libdevice_bc_file})
-      ADD_CUSTOM_COMMAND(
-        OUTPUT "${_libdevice_bc_copy}"
-        DEPENDS "${_libdevice_bc_file}"
-        COMMAND ${CMAKE_COMMAND} -E copy "${_libdevice_bc_file}" "${_libdevice_bc_copy}")
-      LIST(APPEND libdevice_bc ${_libdevice_bc_copy})
-      ADD_DEFINITIONS(-D"__LIBDEVICE_COMPUTE_${libdevice_compute}")
-    ENDIF()
+      FOREACH(cuda_root ${cuda_root_list})
+          SET(_libdevice_bc_file "${cuda_root}/libdevice/libdevice.compute_${libdevice_compute}.10.bc")
+          SET(_libdevice_bc_copy "${PROJECT_BINARY_DIR}/src/backend/cuda/compute_${libdevice_compute}.bc")
+          IF (EXISTS ${_libdevice_bc_file})
+              ADD_CUSTOM_COMMAND(
+                  OUTPUT "${_libdevice_bc_copy}"
+                  DEPENDS "${_libdevice_bc_file}"
+                  COMMAND ${CMAKE_COMMAND} -E copy "${_libdevice_bc_file}" "${_libdevice_bc_copy}")
+              LIST(APPEND libdevice_bc ${_libdevice_bc_copy})
+              ADD_DEFINITIONS(-D"__LIBDEVICE_COMPUTE_${libdevice_compute}")
+              BREAK()
+          ENDIF()
+      ENDFOREACH()
```

Comment:
@pavanky `CUDA_NVVM_HOME` is an internal variable and will not show up in the cache.